### PR TITLE
Refactor table rendering

### PR DESCRIPTION
**Pull request by ChrisJamesC**

### CODE REVIEW

**Comment by wcandillon** [comment]
@ChrisJamesC I tried the PR with the file at https://www.dropbox.com/s/s5n2qmcub2t3qtz/response.json?dl=0
And I get this result:
https://www.dropbox.com/s/ejh5n1tetzgmtxq/Screenshot%202016-12-27%2019.19.43.png?dl=0

Could it be an issue with my setup?

**Comment by ChrisJamesC** [comment]
@wcandillon It was an issue between the chair and the keyboard on my setup. It's fixed now. 
Thanks for providing the example.

**Comment by wcandillon** [comment]
Truly amazing refactoring 👍 
The onclick event on cells doesn't work anymore, I'm investigating.

File: app/directive/layoutmodel.html [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
             <!-- top header row -->
             <th ng-class="colHeaderClasses(header)"
-                ng-click="headerclick({header : header, axis:'x', index: $index, parentIndex: $parent.$parent.$index, facts: table.TableCells.Facts, colspan: header.CellSpan, headerGroup: headerGroup });"
+                ng-click="headerclick({header : header, axis:'x', index: $index, parentIndex: $parent.$parent.$index, facts: table.TableCells.Facts, colspan: header.CellSpan, headerGroup: headerGroup, rows: data });"
```
Comment (ChrisJamesC) [inline app/directive/layoutmodel.html]: `data` is not defined, is it?

**Comment by ChrisJamesC** [comment]
@wcandillon Thanks for the fixes!

**Pull request merged**